vault/resource_aws_secret_backend.go

```diff
@@ -232,141 +276,156 @@ func awsSecretBackendRead(d *schema.ResourceData, meta interface{}) error {
 		// for users of the provider, so to avoid it, we're doing something similar here
 		// and injecting a fake region of us-east-1 into the state.
-		if v, ok := resp.Data["region"].(string); ok && v != "" {
-			d.Set("region", v)
+		if v, ok := resp.Data[consts.FieldRegion].(string); ok && v != "" {
+			d.Set(consts.FieldRegion, v)
 		} else {
-			d.Set("region", "us-east-1")
+			d.Set(consts.FieldRegion, "us-east-1")
 		}
 
-		if v, ok := resp.Data["iam_endpoint"].(string); ok {
-			d.Set("iam_endpoint", v)
-		}
-		if v, ok := resp.Data["sts_endpoint"].(string); ok {
-			d.Set("sts_endpoint", v)
+		for _, k := range awsSecretFields {
+			if v, ok := resp.Data[k]; ok {
+				if err := d.Set(k, v); err != nil {
+					return diag.Errorf("error reading %s for AWS Secret Backend %q: %q", k, path, err)
+				}
+			}
 		}
-		if v, ok := resp.Data["username_template"].(string); ok {
-			d.Set("username_template", v)
+
+		if useAPIVer116 {
+			if err := d.Set(consts.FieldIdentityTokenAudience, resp.Data[consts.FieldIdentityTokenAudience]); err != nil {
+				return diag.Errorf("error reading %s for AWS Secret Backend %q: %q", consts.FieldIdentityTokenAudience, path, err)
+			}
+			if err := d.Set(consts.FieldRoleArn, resp.Data[consts.FieldRoleArn]); err != nil {
+				return diag.Errorf("error reading %s for AWS Secret Backend %q: %q", consts.FieldRoleArn, path, err)
+			}
+			if err := d.Set(consts.FieldIdentityTokenTTL, resp.Data[consts.FieldIdentityTokenTTL]); err != nil {
+				return diag.Errorf("error reading %s for AWS Secret Backend %q: %q", consts.FieldIdentityTokenTTL, path, err)
+			}
 		}
 	}
 
 	d.Set(consts.FieldPath, path)
-	d.Set("description", mount.Description)
-	d.Set("default_lease_ttl_seconds", mount.Config.DefaultLeaseTTL)
-	d.Set("max_lease_ttl_seconds", mount.Config.MaxLeaseTTL)
-	d.Set("local", mount.Local)
+	d.Set(consts.FieldDescription, mount.Description)
+	d.Set(consts.FieldDefaultLeaseTTL, mount.Config.DefaultLeaseTTL)
+	d.Set(consts.FieldMaxLeaseTTL, mount.Config.MaxLeaseTTL)
+	d.Set(consts.FieldLocal, mount.Local)
+	if useAPIVer116 {
+		d.Set(consts.FieldIdentityTokenKey, mount.Config.IdentityTokenKey)
+	}
 
 	return nil
 }
 
-func awsSecretBackendUpdate(d *schema.ResourceData, meta interface{}) error {
+func awsSecretBackendUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client, e := provider.GetClient(d, meta)
 	if e != nil {
-		return e
+		return diag.FromErr(e)
 	}
 
 	path := d.Id()
 	d.Partial(true)
 
 	path, err := util.Remount(d, client, consts.FieldPath, false)
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
-	if d.HasChange("default_lease_ttl_seconds") || d.HasChange("max_lease_ttl_seconds") || d.HasChange("description") {
-		description := d.Get("description").(string)
+	if d.HasChange(consts.FieldDefaultLeaseTTL) || d.HasChange(consts.FieldMaxLeaseTTL) || d.HasChange(consts.FieldDescription) || d.HasChange(consts.FieldIdentityTokenKey) {
```

Comment:
nit: I think we could use `d.HasChanges` here

Reply:
Nice catch! Thank you!